### ISSUE TITLE
GRAVITY_STATE_DIR not respected when galaxy cloned into new location

### ISSUE THREAD

**Issue by jdavcs** [galaxyproject/galaxy#13692]
**Describe the bug**
Installing a fresh clone to a new location (*after having run galaxy from a different location*) and running via run.sh will reuse the value of `GRAVITY_STATE_DIR` from previous installation.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: dev
Commit: 308db49301d486cc0a1294120afb23d8d38bfe6a (fresh dev)

**To Reproduce**
1. never run galaxy
2. clone galaxy into /foo
3. `cd /foo; ./run.sh`
4. logs are in `/foo/database/gravity`
5. clone galaxy into /bar
6. `cd /bar; ./run.sh`
7. logs are still in `foo/database/gravity`; `gravity` dir not created in `bar/database`

**Expected behavior**
Unless I am mistaken, the correct value should be exported in `.venv/bin/activate` (the code is there).

**Comment by nsoranzo** [comment]
@ic4f Cannot reproduce by following your instructions, are you sure you don't have manually set `GRAVITY_STATE_DIR` in your terminal? Since #13685 Galaxy respects that setting if it's exported in the environment.

**Comment by nsoranzo** [comment]
Indeed! I'm not a fan of that hack as well, not sure which could be a better option though.
